KD_Lib/common/base_class.py

```diff
@@ -18,24 +18,37 @@ def __init__(self, teacher_model, student_model, train_loader, val_loader, optim
         self.distl_weight = distil_weight
         self.device = device
 
-    def train_teacher(self, epochs=20, plot_losses=True):
+    def train_teacher(self, epochs=20, plot_losses=True, save_model=True, save_model_pth="./models/teacher.pt"):
         '''
         Function that will be training the teacher 
 
         :param epochs (int): Number of epochs you want to train the teacher 
         :param plot_losses (bool): True if you want to plot the losses
+        :param save_model (bool): True if you want to save the teacher model
+        :param save_model_pth (str): Path where you want to store the teacher model
         '''
```

Comment:
We should also add support to load saved models.

Reply:
yea sure, I'll add this.

Reply:
wait, we don't really need this right. We are initializing our class with this so we won't call `train_teacher()`. This will avoid redundancy. So I don't think we need  to add it here

Reply:
This will require the user to load the weights (from a checkpoint) before creating the BaseClass object (here `attention` object). We okay with this?

Reply:
yes i think we should be okay with this because anyway user is going to initialise the model himself. I guess he will be able to load weights also.